python/whylogs/core/resolvers.py

```diff
@@ -11,11 +12,45 @@
 ColumnSchema: TypeAlias = "ColumnSchema"  # type: ignore
 
 
+AdditionalMetric = namedtuple("AdditionalMetric", "metric column_names why_types")
+
+
 class Resolver(ABC):
     """A resolver maps from a column name and a data type to trackers.
 
     Note that the key of the result dictionaries defines the namespaces of the metrics in the serialized form."""
 
+    def __init__(self):
+        self.additional_metrics: List[AdditionalMetric] = []
+
+    def add_standard_metric(
+        self,
+        metric: StandardMetric,
+        column_names: List[String] = [],
+        why_types: List[Union[Fractional, Integral, String]] = [],
+    ):
+        if not column_names and not why_types:
+            raise ValueError("Either column names or why types must not be empty.")
+        if column_names and why_types:
+            raise ValueError("column_names or why_types should be defined, not both.")
+        if not isinstance(metric, StandardMetric):
+            raise ValueError("Metric must be of StandardMetric type.")
+        additional_metric = AdditionalMetric(metric=metric, column_names=column_names, why_types=why_types)
+        self.additional_metrics.append(additional_metric)
+
+    def resolve_additional_standard_metrics(
+        self, result: Dict[str, Metric], name: str, why_type: DataType, column_schema: ColumnSchema
+    ) -> Dict[str, Metric]:
+        additional_result = {}
+        for additional_metric in self.additional_metrics:
+            for column_name in additional_metric.column_names:
+                if column_name == name and additional_metric.metric.name not in result:
+                    additional_result[additional_metric.metric.name] = additional_metric.metric.zero(column_schema.cfg)
+            for target_why_type in additional_metric.why_types:
+                if isinstance(target_why_type, type(why_type)) and additional_metric.metric.name not in result:
```

Comment:
So this looks like we can only add one ConditionCountMetric? How would I add both a condition count that checks string columns for email pattern, and a condition count that checks the "Fico Score" column is greater than 400? It looks like here the additional_metric.metric.name would just be "condition_count" and limits me to only have one of these?

Reply:
but, right now, there's no easy way to assign different metric instances to different columns, right?

Do we want to change this before merging this PR?

Reply:
With the latest changes, adding different conditions to different columns is easier, by using ResolverSpecs.

There's a test for this case in test suite